### DISCUSSION TITLE
/multilingual-apis-modern-sharepoint/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#451] [Comments]
Using the new multilingual APIs in Modern SharePoint
        
[https://www.eliostruyf.com/multilingual-apis-modern-sharepoint/](https://www.eliostruyf.com/multilingual-apis-modern-sharepoint/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Kalim on 2020-06-18 15:49
```

When you create a translation, I see you have 2 request headers set, but I can't find them in your article. Could you share them with us?

**Comment by comment-migrator** [comment]
```header
Originally posted by jlxq0 on 2020-07-03 08:51
```

Thank you for that great article!  
Do you have any idea if it is possible to connect/assign already existing pages as a translation of another page?

**Reply by comment-migrator** [reply]
```header
Originally posted by Cyrille Grenier on 2021-09-01 14:48
```

a bit old, but did you solve this problem ? I've tried setting the associated \_SP field property using pnp, but doesn't seem to be enough. I suspect that the relation is store in another location, not only on the List Item

**Comment by comment-migrator** [comment]
```header
Originally posted by René on 2020-08-20 18:15
```

How do you know the language of the original page from search? I can retrieve all the translations and know the language of them, but the source doesn’t have this metadata. Or I can’t find it.

**Comment by comment-migrator** [comment]
```header
Originally posted by dd on 2020-09-08 17:20
```

Getting this error for creating translations using api -"The property '\_\_metadata' does not exist on type 'SP.TranslationStatusCreationRequest'. Make sure to only use property names that are defined by the type."

**Comment by comment-migrator** [comment]
```header
Originally posted by Seryoga on 2020-12-01 14:47
```

To resolve the Response Error "The property '\_\_metadata' does not exist on type  
'SP.TranslationStatusCreationRequest'. Make sure to only use property  
names that are defined by the type." you have to add three more headers to the POST-Request:

"content-type": "application/json;odata=verbose;",

"accept": "application/json;odata=verbose;",

"odata-version": "3.0"

**Comment by comment-migrator** [comment]
```header
Originally posted by SharePoint KMDev on 2021-10-13 14:12
```

Hi, thanks for your article.  
I would like to ask a question about a strange behaviour: I create a news page and then its translation (english, then spanish). After publising it, I go to the "see all" summary page (news.aspx) and, when browsing using Spanish content, I see the news page both in Spanish and English. That's not happening when browsing using English content. Have anybody seen this behaviour? Thank you.

**Comment by schneika** [comment]
Due to changes in SharePoint Pages Object Model since last weeks these internal page properties are missing on root language page (in german)
_SPIsTranslation: Specifies if the current page is a translated page
_SPTranslatedLanguages: Specifies which languages are available for the page
_SPTranslationLanguage: The page language
_SPTranslationSourceItemId: The Unique ID of the source page

Do you observe the same behavior?

**Comment by dkouyou** [comment]
Can the translations/create API circumvent the need to publish/approve the source page before creating a translated one?  Because this could help creating translated pages for scheduled pages, something we can't do right now through the UI.
